util/types/json/functions.go

```diff
@@ -65,13 +70,51 @@ func (j JSON) Extract(pathExprList []PathExpression) (ret JSON, found bool) {
 }
 
 // Unquote is for JSON_UNQUOTE.
-func (j JSON) Unquote() string {
+func (j JSON) Unquote() (string, error) {
 	switch j.typeCode {
 	case typeCodeString:
-		return j.str
+		return unquoteString(j.str)
 	default:
-		return j.String()
+		return j.String(), nil
+	}
+}
+
+// unquoteString recognizes the escape sequences shown in:
+// https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#json-unquote-character-escape-sequences
+func unquoteString(s string) (string, error) {
```

Comment:
Need to consider UTF8 characters?

Maybe this function can be used?

https://github.com/pingcap/tidb/blob/26d54c8866b874ee6fe3fd9a3c0841d2d9579939/util/stringutil/string_util.go#L50